Automattic/jetpack
# Search: Fix widget preview with settings

Fixes #22442 

> Updating the Instant Search footer widget and then disabling Instant Search would reset the displaying of the search box and sort selection dropdown on the Classic Search footer widget, which also breaks the widget preview.

The `search box` and `sort selector` would appear on the site only when `search_box_enabled` and `user_sort_enabled` settings are turned on. This change would make the widget preview the same as what is appeared on the site.

<img width="758" alt="截圖 2022-08-23 下午3 26 31" src="https://user-images.githubusercontent.com/6869813/186098447-cb3f85eb-529e-4684-853c-29523abb1afc.png">

<img width="716" alt="截圖 2022-08-23 下午3 26 37" src="https://user-images.githubusercontent.com/6869813/186098477-ce76bbef-2471-4e4d-be27-2eb40bbc40a4.png">

#### Changes proposed in this Pull Request:

* Enable widget preview no matter whether filters or settings set
* Keep widget `search_box_enabled` and `user_sort_enabled` settings when updating on Instant Search mode

#### Other information:


#### Jetpack product discussion


No.

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






1. Purchase a Jetpack Search plan for your site.
2. Ensure that Jetpack Search and Instant Search have been automatically enabled for your site.
3. Add a Jetpack Search widget to your footer (or a sidebar available on all pages).
4. Navigate to `/` and ensure that the widget and the search overlay work as expected.
5. Disable Instant Search.
6. Navigate to `/wp-admin/widgets.php`.
7. Ensure the widget preview exists.

|  Before   | After  |
|  ----  | ----  |
| <img width="809" alt="截圖 2022-08-23 下午3 04 06" src="https://user-images.githubusercontent.com/6869813/186092982-1c8d6bb4-400a-4397-85aa-5684b5108ff2.png"> | <img width="814" alt="截圖 2022-08-23 下午3 05 04" src="https://user-images.githubusercontent.com/6869813/186093016-0e406ef3-4103-477e-bfc3-41df1cd04e47.png"> |

8. Check search widget `search_box_enabled` and `user_sort_enabled` settings and submit the update.
9. Enable Instant Search.
10. Navigate to `/wp-admin/widgets.php`.
11. Change something of the widget and submit the update.
12. Disable Instant Search.
13. Navigate to `/wp-admin/widgets.php`.
14. Ensure the `search_box_enabled` and `user_sort_enabled` settings are kept as checked in step 8.

|  Before   | After  |
|  ----  | ----  |
| <img width="762" alt="截圖 2022-08-23 下午3 09 48" src="https://user-images.githubusercontent.com/6869813/186093957-4786a36d-2659-4ea5-bc66-ffb8a0c1aae5.png"> | <img width="764" alt="截圖 2022-08-23 下午3 10 02" src="https://user-images.githubusercontent.com/6869813/186094013-74e8d0de-8fbb-4c12-bff0-c890127870bb.png"> |
